### PR TITLE
HOLO-1372, HOLO-1374: Add Factory + LayerZeroModule write functions

### PR DESCRIPTION
**Pull request by feliperodriguess**
## Describe Changes

Changes made to the package: **sdk**.

I made it better by doing ...

- Add Factory contract class and its functions (read & write)
- Add LayerZeroModule contract write functions
- Add `@readonly` missing tags 
- Handle read/write on `_getContractFunction` helper for some contract classes
- General polishes

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] All tests are passing

### CODE REVIEW

File: packages/sdk/src/contracts/interfaces.contract.ts [hunk 1]
```diff
@@ -11,11 +11,11 @@ import {Holograph} from './index'
 type HolographInterfacesFunctionNames = ExtractAbiFunctionNames<typeof HolographInterfacesABI>
 
 export enum ChainIdType {
-  UNDEFINED, //  0
-  EVM, //        1
-  HOLOGRAPH, //  2
-  LAYERZERO, //  3
-  HYPERLANE, //   4
+  UNDEFINED, // 0
```
Comment (sogoiii) [inline packages/sdk/src/contracts/interfaces.contract.ts:14]: I think we should have a comment here that links to the solidity code: https://github.com/holographxyz/holograph-protocol/blob/develop/contracts/enum/ChainIdType.sol

File: packages/sdk/src/contracts/interfaces.contract.ts [hunk 2]
```diff
@@ -24,14 +24,14 @@ export enum InterfaceType {
   ERC721, //    2
   ERC1155, //   3
   ROYALTIES, // 4
-  GENERIC, //    5
+  GENERIC, //   5
```
Comment (sogoiii) [inline packages/sdk/src/contracts/interfaces.contract.ts:27]: Add comment that links to solidity code: https://github.com/holographxyz/holograph-protocol/blob/develop/contracts/enum/InterfaceType.sol

File: packages/sdk/src/contracts/interfaces.contract.ts [hunk 3]
```diff
@@ -24,14 +24,14 @@ export enum InterfaceType {
...
 }
 
 export enum TokenUriType {
-  UNDEFINED, //   0
-  IPFS, //        1
-  HTTPS, //       2
-  ARWEAVE, //      3
+  UNDEFINED, // 0
```
Comment (sogoiii) [inline packages/sdk/src/contracts/interfaces.contract.ts:31]: Add comment that links to solidity code: https://github.com/holographxyz/holograph-protocol/blob/develop/contracts/enum/TokenUriType.sol

**Review by BravoNatalie — APPROVED** [review]

**Review by sogoiii — CHANGES_REQUESTED** [review]

**Review by sogoiii — APPROVED** [review]
👍

**Review by alexanderattar — APPROVED** [review]
Looking good!

**Pull request merged**